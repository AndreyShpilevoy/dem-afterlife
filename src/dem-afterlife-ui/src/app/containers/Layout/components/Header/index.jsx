/* eslint fp/no-class: 0, fp/no-nil: 0, fp/no-unused-expression: 0, fp/no-mutation: 0, fp/no-this: 0 */

import React, {PureComponent} from 'react';
import {shape} from 'prop-types';
import {throttle, root, sharedPropTypes, defaults} from 'utils';
import Container from 'components/Container';
import Column from 'components/Column';
import Hidden from 'components/Hidden';
import Row from 'components/Row';
import {css, withStyles} from 'styles';
import Logotype from '../Logotype';
import MenuButton from '../MenuButton';
import NavigationLinkList from '../NavigationLinkList';
import calculateStyles from './calculateStyles';

export class HeaderPure extends PureComponent {
    static propTypes = {
        styles: shape().isRequired,
        navigationLinkArray: sharedPropTypes.navigationLinkArray.isRequired
    };

    constructor(props) {
        super(props);
        this.state = {
            shrinkedHeader: false,
            menuIsClosed: true
        };
    }

    componentDidMount = () => {
        root.addEventListener('scroll', this.handleScroll);
        root.scrollTo(0, 0);
    }

    componentWillUnmount = () => {
        root.removeEventListener('scroll', this.handleScroll);
    }

    handleScroll = throttle(
        ({target}) => {
            target.scrollingElement.scrollTop > 25 ? // eslint-disable-line no-unused-expressions
            this.setState({shrinkedHeader: true}) :
                this.setState({shrinkedHeader: false});
        },
        250);

    handleMenuButtonClick = value => {
        this.setState({menuIsClosed: value});
    }

    render() {
        const {styles, navigationLinkArray} = this.props;
        const {shrinkedHeader, menuIsClosed} = this.state;
        const {header, headerLogoContainer, headerColumn, navigationLinksContainer, fixedOnTheTop,
            headerMenuButtonContainer, navigationLinks, headerPadding} = styles;
        const headerClassName = css([header, shrinkedHeader ? 'shrinkedHeader' : defaults.emptyString]);
        const headerLogoContainerClassName = css([headerLogoContainer, shrinkedHeader ? 'shrinkedHeader' : defaults.emptyString]);
        const navigationLinksContainerClassName = css([headerColumn, navigationLinksContainer, menuIsClosed ? 'closed' : defaults.emptyString]);
        const menuButtonOnClick = () => this.handleMenuButtonClick(!menuIsClosed);
        return (
            <div>
                <Container className={fixedOnTheTop}>
                    <div className={headerClassName}>
                        <Row>
                            <Column xs={7} sm={4} xl={3} className={headerColumn}>
                                <Logotype className={headerLogoContainerClassName} />
                            </Column>
                            <Column xs={2} xsOffset={3} sm={1} smOffset={7} lg={0} lgOffset={0} className={headerColumn}>
                                <Hidden lg={'up'} className={headerMenuButtonContainer}>
                                    <MenuButton onClick={menuButtonOnClick} />
                                </Hidden>
                            </Column>
                            <Column xs={12} lg={8} xl={9} className={navigationLinksContainerClassName}>
                                <NavigationLinkList navigationLinkArray={navigationLinkArray} className={navigationLinks} />
                            </Column>
                        </Row>
                    </div>
                </Container>
                <div className={headerPadding} />
            </div>
        );
    }
}

export default withStyles(theme => calculateStyles(theme), {pureComponent: true})(HeaderPure);