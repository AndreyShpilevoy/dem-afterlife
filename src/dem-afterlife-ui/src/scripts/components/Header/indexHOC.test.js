/* eslint no-undef: 0, fp/no-unused-expression: 0, fp/no-nil: 0*/

import React from 'react';
import {shallow} from 'enzyme';
import Header from './index';

describe('Header HOC', () => {
    const hocProps = {
        theme: 'default'
    };

    it('component match expected snapshot', () => {
        expect(shallow(<Header theme={hocProps.theme} navigationLinkArray={[]}/>)).toMatchSnapshot();
    });
});
