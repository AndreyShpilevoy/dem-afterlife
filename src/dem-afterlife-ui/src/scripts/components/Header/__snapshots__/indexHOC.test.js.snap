// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header HOC component match expected snapshot 1`] = `
<HeaderPure
  classNames={
    Object {
      ".header-0-2.shrinkedHeader": "header-0-2",
      "fixedOnTheTop": "fixedOnTheTop-0-0",
      "header": "header-0-2",
      "headerPadding": "headerPadding-0-3",
      "logotypeColumn": "logotypeColumn-0-1",
    }
  }
  theme="default"
/>
`;
